reflex/components/markdown/markdown.py

```diff
@@ -239,43 +344,50 @@ def format_component(self, tag: str, **props) -> str:
         """
         return str(self.get_component(tag, **props)).replace("\n", "")
 
-    def format_component_map(self) -> dict[str, Var]:
-        """Format the component map for rendering.
+    def _get_map_fn_var_from_children(self, component: Component, tag: str) -> Var:
+        """Create a function Var for the component map for the specified tag.
+
+        Args:
+            component: The component to check for custom code.
+            tag: The tag of the component.
 
         Returns:
-            The formatted component map.
+            The function Var for the component map.
         """
-        components = {
-            tag: Var(
-                _js_expr=f"(({{node, {_CHILDREN._js_expr}, {_PROPS._js_expr}}}) => ({self.format_component(tag)}))"
-            )
-            for tag in self.component_map
-        }
+        if isinstance(component, MarkdownComponentMap):
+            return component.create_map_fn_var(f"({self.format_component(tag)})")
 
-        # Separate out inline code and code blocks.
-        components["code"] = Var(
-            _js_expr=f"""(({{node, inline, className, {_CHILDREN._js_expr}, {_PROPS._js_expr}}}) => {{
-    const match = (className || '').match(/language-(?<lang>.*)/);
-    const language = match ? match[1] : '';
-    if (language) {{
-    (async () => {{
-      try {{
-        const module = await import(`react-syntax-highlighter/dist/cjs/languages/prism/${{language}}`);
-        SyntaxHighlighter.registerLanguage(language, module.default);
-      }} catch (error) {{
-        console.error(`Error importing language module for ${{language}}:`, error);
-      }}
-    }})();
-  }}
-    return inline ? (
-        {self.format_component("code")}
-    ) : (
-        {self.format_component("codeblock", language=Var(_js_expr="language", _var_type=str))}
-    );
-      }})""".replace("\n", " ")
-        )
+        # fallback to the default fn Var creation if the component is not a MarkdownComponentMap.
+        return MarkdownComponentMap.create_map_fn_var(f"({self.format_component(tag)})")
```

Comment:
The behavior here is to get the map_fn_var of the component. However just like the existing behaviour was (before this PR), it is--and was--assumed that the targetted component is not wrapped in a parent component or the component is not memorized (using `rx.memo`). This assumption is a bit problematic.
Eg.
```python
component_map = {
"p": lambda value: Text.create(value, margin_y="1em"),
}
```
If the component in the lambda fn (Text) mixes `MarkdownComponentMap`, it is straightforward to obtain the map_fn_var. However, if the component returns  the component in a wrapper (eg. `Text.create` returning `Text` wrapped in a `Box`), then its not obvious how to extract the var. Another eg is the case of returning a memorized fn that returns a component instead of a direct component:

```python
@rx.memo
def code_block(code: str, language: str):
    return rx.box(
        rx.code_block(
            code,
            language=language,
            class_name="code-block",
            can_copy=True,
        ),
        class_name="relative mb-4",
    )

def code_block_markdown(*children, **props):
    language = props.get("language", "plain")
    return code_block(code=children[0], language=language)

...
rx.markdown("# header", component_map={"codeblock": code_block_markdown}),
```
(`codeblock` is treated differently because of the inline logic we have so this should be fine but think of an alternative eg)

A fix for this was to recursively traverse the component tree and find the child with the appropriate var, but this introduces another problem where a parent component is of the `MarkdownComponentMap` type. In this case, what do we do? I think this is one to think about or consider. I initially had this implementation as a recursive one but I had to revert this to follow the exactly behavior as was previously